pkg/bgpv1/manager/manager.go

```diff
@@ -1045,13 +1079,21 @@ func getRouterID(config *v2alpha1api.CiliumBGPNodeInstance, ciliumNode *v2api.Ci
 	if err != nil {
 		return "", fmt.Errorf("failed to parse Node annotations for instance %v: %w", config.Name, err)
 	}
-
 	routerID, err := annoMap.ResolveRouterID(asn)
+
+	// If there are no annotations about router-id, router-id will be allocated based on the allocation mode
 	if err != nil {
-		if nodeIP := ciliumNode.GetIP(false); nodeIP == nil {
-			return "", fmt.Errorf("failed to get CiliumNode IP %v: %w", nodeIP, err)
-		} else {
+		// check if the router ID allocation mode is set to default
+		isDefault := option.Config.BGPRouterIDAllocationMode == defaults.BGPRouterIDAllocationMode
+		if !isDefault {
+			// TODO: Implement other router ID allocation modes
+			return "", fmt.Errorf("Allocation methods other than default are not supported: %w", err)
+		} else if nodeIP := ciliumNode.GetIP(false); nodeIP != nil {
 			routerID = nodeIP.String()
+		} else if routerIDFromMacString, _ := calcRouterIDFromMacAddress(); routerIDFromMacString != "" {
```

Comment:
Is it intended to not use the error generated by `calcRouterIDFromMacAddress()` ?